pkg/util/enforcement_action.go

```diff
@@ -30,14 +55,79 @@ var ErrEnforcementAction = errors.New("unrecognized enforcementAction")
 // spec.enforcementAction field as it was not a string.
 var ErrInvalidSpecEnforcementAction = errors.New("spec.enforcementAction must be a string")
 
-func ValidateEnforcementAction(input EnforcementAction) error {
-	for _, n := range supportedEnforcementActions {
-		if input == n {
-			return nil
+var ErrUnrecognizedEnforcementPoint = errors.New("unrecognized enforcement points")
+
+var ErrInvalidSpecScopedEnforcementAction = errors.New("spec.scopedEnforcementAction must be in the format of []{action: string, enforcementPoints: []{name: string}}")
+
+func ValidateEnforcementAction(input EnforcementAction, item map[string]interface{}) error {
+	switch input {
+	case Scoped:
+		return ValidateScopedEnforcementAction(item)
+	case Dryrun, Deny, Warn:
+		return nil
+	default:
+		return fmt.Errorf("%w: %q is not within the supported list %v",
+			ErrEnforcementAction, input, supportedEnforcementActions)
+	}
+}
+
+func ValidateScopedEnforcementAction(item map[string]interface{}) error {
+	obj, err := GetScopedEnforcementAction(item)
+	if err != nil {
+		return fmt.Errorf("error fetching scopedEnforcementActions: %w", err)
+	}
+
+	var unrecognizedEnforcementPoints []string
+
+	// validating scopedEnforcementActions
+	for _, scopedEnforcementAction := range *obj {
+		if scopedEnforcementAction.Action == "" {
+			return ErrInvalidSpecEnforcementAction
+		}
+		switch EnforcementAction(scopedEnforcementAction.Action) {
+		case Dryrun, Deny, Warn:
+		default:
+			return fmt.Errorf("%w: %q is not within the supported list %v", ErrEnforcementAction, scopedEnforcementAction.Action, supportedScopedActions)
+		}
+		if len(scopedEnforcementAction.EnforcementPoints) == 0 {
+			return ErrInvalidSpecEnforcementAction
+		}
+		for _, enforcementPoint := range scopedEnforcementAction.EnforcementPoints {
+			switch strings.ToLower(enforcementPoint.Name) {
```

Comment:
Will remove this if needed after we reach to a conclusion on - https://github.com/open-policy-agent/frameworks/pull/403#discussion_r1695885749